source/common/upstream/load_balancer_impl.cc

```diff
@@ -12,6 +12,21 @@
 
 namespace Envoy {
 namespace Upstream {
+namespace {
+
+const HostSet* bestAvailable(const PrioritySet* priority_set) {
+  if (priority_set == nullptr) {
+    return nullptr;
+  }
+  for (auto& host_set : priority_set->hostSetsPerPriority()) {
+    if (!host_set->healthyHosts().empty()) {
+      return host_set.get();
+    }
+  }
+  return priority_set->hostSetsPerPriority()[0].get();
```

Comment:
Can `priority_set->hostSetsPerPriority().size() == 0` here ever?

Reply:
No, we always populate the first priority set.

Reply:
+1 adding assert/comment here about this.

Reply:
Still need assert/comment here I think?